### ISSUE TITLE
Cannot define a string inside a custom column

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#34776]
### Describe the bug

Not able to define a string in a custom column. This issues causes 3 bugs in and i will open on them below

### To Reproduce

1. Go to New Question -> Custom Column -> `"Total"` -> See error message

<img width="1459" alt="image" src="https://github.com/metabase/metabase/assets/110378427/e18bcb26-1aae-46dd-b2e2-f3999d36a7d5">

```
Input to resolve-card-id-source-tables* does not match schema: [0;33m [(named {:query {:expressions {"total" (named (not (some-matching-condition? "Total")) "Must be a valid instance of one of these clauses: :expression, :field")}}} arg0)] [0m
```
2. You can try to bypass this by using `concat("Total")` -> But the column simply doesn't show:

<img width="1179" alt="image" src="https://github.com/metabase/metabase/assets/110378427/bc6617c3-b0cb-4574-9e5b-8d7857a4e266">

3. Finally you can use `concat("Total","")` and it works:

<img width="1233" alt="image" src="https://github.com/metabase/metabase/assets/110378427/4649306a-11f6-4917-a0c1-ca2e74eb84f2">

BUT the moment you use this column as an aggregate it fails again:

<img width="1498" alt="image" src="https://github.com/metabase/metabase/assets/110378427/678401d4-c051-4345-998e-3c7b59edd3ca">

```
Column "source.total_2" not found; SQL statement: SELECT "source"."total_2" AS "total_2", COUNT(*) AS "count" FROM (SELECT CONCAT(?, ?) AS "total" FROM "PUBLIC"."ORDERS" LEFT JOIN "PUBLIC"."PRODUCTS" AS "PRODUCTS__via__PRODUCT_ID" ON "PUBLIC"."ORDERS"."PRODUCT_ID" = "PRODUCTS__via__PRODUCT_ID"."ID" LEFT JOIN "PUBLIC"."PEOPLE" AS "PEOPLE__via__USER_ID" ON "PUBLIC"."ORDERS"."USER_ID" = "PEOPLE__via__USER_ID"."ID") AS "source" GROUP BY "source"."total_2" ORDER BY "source"."total_2" ASC LIMIT 10 [42122-214]

```

### Expected behavior

These 3 bugs shouldn't occur when the result of a custom column is text

### Logs

Error message shared above there are 3 different error messages depending on which stage you hit

### Information about your Metabase installation

```JSON
1.47.4 and master
```

### Severity

This limits the GUI functionality when working with strings and due to the nature of the bug it makes the whole experience pretty frustrating

### Additional context

In step 2 if the name of the column doesn't reference a column in the table. Something like test as the name of the custom column then step 2 and 3 work given that you use the `concat("Total","")`  hack.

<img width="1505" alt="image" src="https://github.com/metabase/metabase/assets/110378427/150946b9-db5d-4c09-8265-7f1d8d65880f">

**Comment by luizarakaki** [comment]
@Tony-metabase I changed some labels here. Did you find any FE root cause? These errors come from the BE

**Comment by Tony-metabase** [comment]
You are right @luizarakaki! Sorry was a mistake from my end ... This is a BE issue

**Comment by ranquild** [comment]
using constants is a new feature; bug other points are bugs

**Comment by paoliniluis** [comment]
@Tony-metabase this is not possible anymore, can you check?

![Image](https://github.com/user-attachments/assets/a814fe30-7b56-4e48-beb4-7c11adce7f69)
